media/filters/chunk_demuxer.cc

```diff
@@ -490,6 +500,13 @@ void ChunkDemuxerStream::CompletePendingReadIfPossible_Locked() {
   // Other cases are kOk and just return the buffers.
   DCHECK(!buffers.empty());
   requested_buffer_count_ = 0;
+#if BUILDFLAG(USE_STARBOARD_MEDIA)
+  for (auto&& buffer : buffers) {
+    if (!buffer->end_of_stream()) {
+      write_head_ = std::max(write_head_, buffer->timestamp());
```

Comment:
Did we consider extending media/filters/chunk_demuxer_unittest.cc for the new behaviour?